backend/manager/modules/bll/src/main/java/org/ovirt/engine/core/bll/storage/disk/MoveOrCopyDiskCommand.java

```diff
@@ -277,19 +277,19 @@ protected boolean validateSpaceRequirements() {
             sdsToValidate.add(getParameters().getSourceDomainId());
         }
 
-        MultipleStorageDomainsValidator storageDomainsValidator =
-                createMultipleStorageDomainsValidator(sdsToValidate);
+        MultipleStorageDomainsValidator storageDomainsValidator = createMultipleStorageDomainsValidator(sdsToValidate);
 
-        if (validate(storageDomainsValidator.allDomainsWithinThresholds())) {
-            // If we are copying a template's disk we do not want all its copies
-            if (getImage().getVmEntityType() == VmEntityType.TEMPLATE) {
-                getImage().getSnapshots().add(getImage());
-            } else {
-                getImage().getSnapshots().addAll(diskImageDao.getAllSnapshotsForLeaf(getImage().getImageId()));
-            }
-            return validate(storageDomainsValidator.allTargetDomainsHaveSpaceForCopiedDisks(Collections.singletonList(getImage())));
+        if (!validate(storageDomainsValidator.allDomainsWithinThresholds())) {
+            return false;
         }
-        return false;
+
+        // If we are copying a template's disk we do not want all its copies
+        if (getImage().getVmEntityType() == VmEntityType.TEMPLATE) {
```

Comment:
Can `isTemplate()` be used here?

Reply:
I'm not sure how we should treat the case there's no entity type set